### PR TITLE
Remove systemd package from kubekins bootstrap image

### LINKED ISSUE #7844: Remove systemd from kubekins/bootstrap image
Running systemd related binaries in the bootstrap image for kubekins clearly fails.

```
dims@bigbox:~$ docker run --privileged -it --rm --entrypoint=/bin/bash gcr.io/k8s-testimages/bootstrap:v20180410-dcf7bea8a -c /bin/bash
root@7a29b8dafe5d:/workspace# which systemd-run
/usr/bin/systemd-run
root@7a29b8dafe5d:/workspace# systemctl status
Failed to get D-Bus connection: Unknown error -1
root@7a29b8dafe5d:/workspace# systemd-run env
Failed to create bus connection: No such file or directory
root@7a29b8dafe5d:/workspace# dpkg -l | grep systemd
ii  libsystemd0:amd64           215-17+deb8u7                    amd64        systemd utility library
ii  systemd                     215-17+deb8u7                    amd64        system and service manager
ii  systemd-sysv                215-17+deb8u7                    amd64        system and service manager - SysV links
```

We are clearly not using systemd and we are using upstart
```
root@7a29b8dafe5d:/workspace# service --status-all
 [ - ]  bootlogs
 [ - ]  bootmisc.sh
 [ - ]  checkfs.sh
 [ - ]  checkroot-bootclean.sh
 [ - ]  checkroot.sh
 [ - ]  docker
 [ - ]  hostname.sh
 [ ? ]  hwclock.sh
 [ - ]  killprocs
 [ - ]  motd
 [ - ]  mountall-bootclean.sh
 [ - ]  mountall.sh
 [ - ]  mountdevsubfs.sh
 [ - ]  mountkernfs.sh
 [ - ]  mountnfs-bootclean.sh
 [ - ]  mountnfs.sh
 [ - ]  procps
 [ - ]  rc.local
 [ - ]  rmnologin
 [ - ]  rsync
 [ - ]  sendsigs
 [ + ]  udev
 [ ? ]  udev-finish
 [ - ]  umountfs
 [ - ]  umountnfs.sh
 [ - ]  umountroot
 [ - ]  unattended-upgrades
 [ - ]  urandom
```

So we should remove the systemd package from the image

### PR DESCRIPTION
**Pull request by dims**
Fixes #7844

### CODE REVIEW

File: images/bootstrap/Dockerfile [hunk 1]
```diff
@@ -116,6 +116,8 @@ ADD ["./barnacle/barnacle", "/usr/local/bin"]
 # END: DOCKER IN DOCKER SETUP
 #
 
+# Cleanup unnecessary package(s)
+RUN apt-get remove -y systemd
```
Comment (BenTheElder) [inline images/bootstrap/Dockerfile]: this should be early in the image since it's unlikely to change any more often than the base image.

Comment (dims) [reply images/bootstrap/Dockerfile]: Moved it as early as i can. any earlier and it fails with the following

```
The following packages will be REMOVED:
  init systemd systemd-sysv
WARNING: The following essential packages will be removed.
This should NOT be done unless you know exactly what you are doing!
  init systemd-sysv (due to init)
0 upgraded, 0 newly installed, 3 to remove and 0 not upgraded.
After this operation, 11.8 MB disk space will be freed.
E: There are problems and -y was used without --force-yes
The command '/bin/sh -c apt-get remove -y systemd' returned a non-zero code: 100
make: *** [build] Error 100
```

File: images/bootstrap/Dockerfile [hunk 2]
```diff
@@ -50,6 +50,9 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
     && apt-get clean \
     && python -m pip install --upgrade pip setuptools wheel
 
+# Cleanup unnecessary package(s)
+RUN apt-get remove -y systemd
```
Comment (BenTheElder) [inline images/bootstrap/Dockerfile]: is there any reason to have this after we install things? seems like this should be before ARG IMAGE_ARG even.

Comment (dims) [reply images/bootstrap/Dockerfile]: @BenTheElder if i move it any earlier the command fails ... i tried it

Comment (BenTheElder) [reply images/bootstrap/Dockerfile]: ?? is something depending on systemd then? what's the failure ??

Comment (dims) [reply images/bootstrap/Dockerfile]: ok figured it out . needed special incantation - https://superuser.com/questions/921354/script-answer-to-apt-get-request-to-continue-type-in-the-phrase-yes-do-as-i-s

**Comment by dims** [comment]
@BenTheElder this is ready when you get a chance ( no rush! )

File: images/bootstrap/Dockerfile [hunk 3]
```diff
@@ -22,6 +22,9 @@ RUN mkdir -p /workspace
 ENV WORKSPACE=/workspace \
     TERM=xterm
 
+# Cleanup unnecessary package(s)
+RUN echo 'Yes, do as I say!' | apt-get remove systemd
```
Comment (BenTheElder) [inline images/bootstrap/Dockerfile]: this is pretty much the best line of bash in `test-infra`.

**Comment by BenTheElder** [comment]
I want to check this out locally and inspect the built image, will do that soon.

File: images/bootstrap/Dockerfile (hunk 3 above)
Comment (BenTheElder) [reply images/bootstrap/Dockerfile]: I'm moderately certain now that something below depends on systemd. If I `apt list --installed` in the built image I see:
```
systemd/oldstable,now 215-17+deb8u7 amd64 [installed,automatic]
systemd-sysv/oldstable,now 215-17+deb8u7 amd64 [installed,automatic]
```

**Comment by BenTheElder** [comment]
This totally slipped, just took a look. Sorry about that.

**Comment by dims** [comment]
@BenTheElder ping :)

**Comment by dims** [comment]
ack. will leave the hack i have in the scripts as-is.

/close

**Comment by BenTheElder** [comment]
I want to investigate this further but have a few other priorities, imo we should be stripping everything we don't actually use out of the images anyhow...

**Pull request closed without merging**